src/metalperformanceshaders.cs

```diff
@@ -710,15 +710,35 @@ interface MPSCnnKernel {
 		[Export ("clipRect", ArgumentSemantic.Assign)]
 		MTLRegion ClipRect { get; set; }
 
+		[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
 		[Export ("destinationFeatureChannelOffset")]
 		nuint DestinationFeatureChannelOffset { get; set; }
 
+		[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+		[Export ("sourceFeatureChannelMaxCount")]
+		nuint SourceFeatureChannelMaxCount { get; set; }
+
+		[Export ("sourceFeatureChannelOffset")]
+		nuint SourceFeatureChannelOffset { get; set; }
+
 		[Export ("edgeMode", ArgumentSemantic.Assign)]
 		MPSImageEdgeMode EdgeMode { get; set; }
 
 		[Export ("encodeToCommandBuffer:sourceImage:destinationImage:")]
 		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSImage sourceImage, MPSImage destinationImage);
 
+		[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)] // keeping same name to match existing member
+		[Export ("encodeToCommandBuffer:sourceImage:destinationState:destinationImage:")]
+		void EncodeToCommandBuffer (IMTLCommandBuffer commandBuffer, MPSImage sourceImage, MPSState destinationState, MPSImage destinationImage);
+
+		[TV (11,3), Mac (10,13,4, onlyOn64: true), iOS (11,3)]
+		[Export ("encodeBatchToCommandBuffer:sourceImages:destinationImages:")]
+		void EncodeBatch (IMTLCommandBuffer commandBuffer, NSArray<MPSImage> sourceImages, NSArray<MPSImage> destinationImages);
```

Comment:
My reasoning behind using `NSArray<T>` in MPS is because both `NSArray<MPSImage>` and `NSArray<MPSState>` are extensively used trough the API and it seems that they will be reused several times so instead of going back and forth between managed array and `NSArray` on each API I figured just to keep everything in NSArray, also sometimes they are used as an out container, meaning that you just give it an `NSArray` of a fixed size and the API will fill its contents.